### ISSUE TITLE
fix[dev]: 🤔 make `sd-step(-group)` accessible

### ISSUE THREAD

**Issue by karlbaumhauer** [solid-design-system/solid#1171]
## Description
This tasks describes the Figma implementation for: solid-design-system/internal#78 
Chapter 9.1.3.2 in the file "Test Details"

### Task
Currently `sd-divider` is used in `sd-step-group` for the tail. This uses wrong semantics under the hood (`hr`). This should be refactored to use simple `divs` with a border, which would simplify CSS as well.

### Open Questions towards design
- [ ] Question1
- [ ] Question2

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

**Comment by mariohamann** [comment]
I started a draft PR with the refactoring, but had problems with the vertical version. @Vahid1919 could you please have a look?

https://github.com/solid-design-system/solid/pull/1390
